service/storage_proxy.cc

```diff
@@ -5218,14 +5218,23 @@ storage_proxy::query_partition_key_range_concurrent(storage_proxy::clock_type::t
                 break;
             } else if (pcf) {
                 // check that merged set hit rate is not to low
-                auto find_min = [&g = _remote->gossiper(), pcf] (const inet_address_vector_replica_set& range) {
+                auto find_min = [this, pcf] (const inet_address_vector_replica_set& range) {
+                    if (range.size() == 1 && range[0] == utils::fb_utilities::get_broadcast_address()) {
```

Comment:
Again, I was expecting a check against the replication strategy of the table queried, instead of looking at how many nodes we have. Can we enter this branch in a degraded cluster, and happily proceed with a query instead of erroring out?

Reply:
This refactor should not change the logic (if it does, I made a mistake). The result of `find_min` should be the same before and after the refactor in all possible states / for all possible arguments. Indeed, this is very easy to verify for `find_min`: it's just a call to `boost::min_element`, which will always return the only member of `range` if `range.size() == 1`.

Reply:
ok